nativelib/src/main/scala/scala/scalanative/runtime/CMemoryPool.scala

```diff
@@ -0,0 +1,38 @@
+package scala.scalanative.runtime
+
+import scalanative.unsafe._
+
+@extern
+object _CMemoryPool {
+  @name("memorypool_open")
+  def open(): RawPtr = extern
+
+  @name("memorypool_free")
+  def free(rawpool: RawPtr): Unit = extern
+}
+
+object CMemoryPool {
+  val defaultMemoryPoolHandle = _CMemoryPool.open() // TODO: use lazy val
+
+  def freeDefaultMemoryPool(): Unit = {
+    _CMemoryPool.free(defaultMemoryPoolHandle)
+  }
+}
```

Comment:
You can probably make the extern bindings an private inner object, this way we can make it a bit more cleaner and prevent leaking the access to the internals

Also, we should be able to see lazy val safely here, but it does not make a big difference - it would be allocated on the first usage anyway.

```suggestion
object CMemoryPool {
  lazy val defaultMemoryPoolHandle = Impl.open() // TODO: use lazy val

  def freeDefaultMemoryPool(): Unit =  Impl.free(defaultMemoryPoolHandle)
  
  @extern
  private object Impl {
    @name("memorypool_open")
    def open(): RawPtr = extern

    @name("memorypool_free")
    def free(rawpool: RawPtr): Unit = extern
  }
}
```